sdk/core/core-amqp/src/retry.ts

```diff
@@ -26,6 +26,27 @@ function isDelivery(obj: any): boolean {
   return result;
 }
 
+/**
+ * TODO: We should question whether it's even a legitimate way of setting the timeout
```

Comment:
A faithful port, even of the TODO comment questioning the legitimacy of the function. ;)